### PR TITLE
adol-c: add variant stdczero

### PR DESCRIPTION
**Pull request by severinstrobl**

### CODE REVIEW

File: var/spack/repos/builtin/packages/adol-c/package.py [hunk 1]
```diff
@@ -80,6 +82,11 @@ def configure_args(self):
                 '--enable-atrig-erf'
             )
 
+        if '~stdczero' in spec:
+            configure_args.append(
+                '--disable-stdczero'
+            )
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/adol-c/package.py]:
```suggestion
        configure_args.extend(self.enable_or_disable('stdczero'))
```

File: var/spack/repos/builtin/packages/adol-c/package.py [hunk 2]
```diff
@@ -30,6 +30,8 @@ class AdolC(AutotoolsPackage):
             description='Enable advanced branching to reduce retaping')
     variant('atrig_erf', default=True,
             description='Enable arc-trig and error functions')
+    variant('stdczero', default=True,
+            description='Enable default initialization for the adouble datatype')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/adol-c/package.py:34]: Is this a typo:
```suggestion
            description='Enable default initialization for the double datatype')
```
or there is an `adouble` datatype?

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/adol-c/package.py (hunk 1 above)
Comment (severinstrobl) [reply var/spack/repos/builtin/packages/adol-c/package.py]: Should I unify the handling of the existing configure args while I'm at it?

File: var/spack/repos/builtin/packages/adol-c/package.py (hunk 2 above)
Comment (severinstrobl) [reply var/spack/repos/builtin/packages/adol-c/package.py:34]: The type is actually `adouble`, which is the AD version of a plain `double`.

File: var/spack/repos/builtin/packages/adol-c/package.py (hunk 1 above)
Comment (alalazo) [reply var/spack/repos/builtin/packages/adol-c/package.py]: @severinstrobl If you want to that would be great!

**Review by alalazo — APPROVED** [review]

**Pull request merged**